tests/test_dump_file.py

```diff
@@ -8,18 +8,16 @@
 
 @pytest.mark.parametrize("sort_ids", [False, True])
 @pytest.mark.parametrize("shuffle_ids", [False, True])
-@pytest.mark.parametrize("use_gzip", [False, True])
-def test_dump_file_min(snap, use_gzip, shuffle_ids, sort_ids, tmp_path):
+@pytest.mark.parametrize("compression_extensions", ["", ".gz", ".zstd"])
+def test_dump_file_min(snap, compression_extensions, shuffle_ids, sort_ids, tmp_path):
```

Comment:
A few things here:

1. `compression_extension` should be singular because there is only one tested at a time.
2. You have the wrong file extension for zstd, so the test isn't actually doing what you think it is! The correct extension is `.zst` based on the code above and a Google search.
3. These tests should probably be skipped if pyzstd is not installed.
4. We should then make sure to configure the tests to install pyzstd and make sure these run.

The small fixes needed here are:
```suggestion
@pytest.mark.parametrize("compression_extension", ["", ".gz", ".zst"])
def test_dump_file_min(snap, compression_extension, shuffle_ids, sort_ids, tmp_path):
```

Additionally, I suggest that at the top of the file, you do:
```py
try:
    import pyzstd
    has_pyzstd = True
except ModuleNotFoundError:
    has_pyzstd = False
```
Then in any tests that test zstd compression, add the following right at the beginning to conditionally skip:
```py
if not has_pyzstd and compression_extension == ".zst":
    pytest.skip("pyzstd not installed")
```

Discuss with me about how to install `pyzstd` in the testing environment, as we have two options there.